spec/functional/resource/dpkg_package_spec.rb

```diff
@@ -45,11 +45,19 @@ def set_dpkg_package_name(name)
     dpkg_package.package_name name
   end
 
-  def should_be_purged_or_removed(package, action)
+  def should_be_purged_or_removed(package, action=nil)
+    status = shell_out("dpkg -s #{package}")
+    output = status.stdout + status.stderr
     if action == :purge
-      shell_out!("dpkg -s #{package}", returns: [1])
+      not_installed = !(output =~ /deinstall/) &&
+        (!!(output =~ /not installed/) || !!(output =~ /not-installed/))
```

Comment:
i'd actually say the `!!(foo =~ /bar/)`'s here are not improving readability.  even though =~ doesn't return a boolean the || will sort that out just fine.